bpf-gpl/tc.c

```diff
@@ -455,6 +455,25 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 		}
 	}
 
+	// Drop malformed IP packets
+	if (ip_header->ihl < 5) {
+		fwd.reason = CALI_REASON_IP_MALFORMED;
+		CALI_DEBUG("Drop malformed IP packets\n");
+		goto deny;
+	}
+	else if (ip_header->ihl > 5) {
+		/* Drop packets with IP options from/to WEP.
+		 * Also drop packets with IP options if the dest IP is not host IP
+		 */
+		if (CALI_F_FROM_WEP || CALI_F_TO_WEP ||
+			!cali_rt_flags_local_host(cali_rt_lookup_flags(ip_header->daddr))) {
```

Comment:
I think you should only check the dest IP in the FROM_HEP program.  TO_HEP is egress from the host so it'd be the source IP that'd be relevant there.

No need to check the source IP in TO_HEP since either:
* it's from the host so we should allow
* it's from a workload, but then we'd have dropped it already in the FROM_WEP hook.